### DISCUSSION TITLE
How to render different fonts with draw_cad.py?

### DISCUSSION THREAD

**Discussion by ru4ert** [mozman/ezdxf#1002] [Q&A] [answered]
When i exe`python ./draw_cad.py T1704180658.dxf  --out image.png` i get 3 times the same font:
![T1704180658](https://github.com/mozman/ezdxf/assets/47078678/2698eb59-db9c-4aa2-b605-73a2650297b9)

but it should be:

![image](https://github.com/mozman/ezdxf/assets/47078678/71ed1c23-34ac-45b7-9f6d-7d2993b6abcc)

T1704180658.dxf file: https://gist.githubusercontent.com/ru4ert/0f099e0ad5133b36547b5cb6663ed4c5/raw/e7b1c1e1ccba7b7004d6c21c8dcecc04814f0c51/gistfile1.txt

ezdxf 1.1.4
matplotlib 3.8.2

**Comment by mozman** [comment]
First always rebuild the font-manager cache: https://ezdxf.mozman.at/docs/howto/fonts.html#

I guess this DXF file was created on macOS.

Because the TTF fonts in the DXF file are named differently than on Windows:

- `Comic Sans MS.ttf`  is `comic.ttf` on Windows
- `Arial Black.ttf` is `ariblk.ttf` on Windows
...

DWG TrueView (AutoCAD) does an auto-translation for this font names on Windows:

![image](https://github.com/mozman/ezdxf/assets/4230966/16d74e91-7fca-4dc4-85df-d7a587ae8722)

BricsCAD doesn't do this:

![image](https://github.com/mozman/ezdxf/assets/4230966/ea413d56-ae34-42cb-a99e-195df374a21f)

and `ezdxf` also doesn't do this.

If you open this DXF file on a mac the fonts should be rendered as expected - except the fonts are not stored where `ezdxf` is looking for.

This is the only location on macOS:
```Python
MACOS_FONT_DIRS = ["/Library/Fonts/"]
```

You can add more font directories to the config file at key `support_dirs`:

```ini
[core]
default_dimension_text_style = OpenSansCondensed-Light
test_files = C:\Source\dxftest
support_dirs =
        C:\Source\ezdxf.git\fonts\shapefiles\
        "another font directory"
        "and so on"
```
**... and rebuild the font-manager cache.**

More about config files: https://ezdxf.mozman.at/docs/options.html#config-files

Please post other default font folders if there some and I will add them to the default search locations for macOS.

FYI: you can attach DXF files to a post in a zip-archive

**Reply by mozman** [reply, marked as answer]
I added support for font-name synonyms.

The DXF file with macOS font-names on Windows:

![image](https://github.com/mozman/ezdxf/assets/4230966/04f9c3fc-af04-4049-a306-3020e2b9d47a)

I don't have a Mac and never will. Therefore this feature is only as good as users contribute to the [font_synonyms.py](https://github.com/mozman/ezdxf/blob/master/src/ezdxf/fonts/font_synonyms.py) file, which contains only the font-names of your sample file yet:
```Python
FONT_SYNONYMS = {
    "ariblk.ttf": "Arial Black.ttf",
    "comic.ttf": "Comic Sans MS.ttf",
    "arialuni.ttf": "Arial Unicode.ttf",
    "times.ttf": "Times New Roman.ttf",
    "trebuc.ttf": "Trebuchet MS.ttf",   
}
```

**Reply by ru4ert** [reply]
for me, working on mac i created a 

**ezdxf.ini**
```
[core]
support_dirs =
        "/System/Library/Fonts/"
        "/Library/Fonts/"
```

**Comment by ru4ert** [comment]
@mozman thank you. I created a PR for an additional font dir on mac. #1004

**Comment by ru4ert** [comment]
Good to know, the dxf was created on mac as well the dxf to image is also on mac
